### ISSUE TITLE
TypeError: '_PropertyDeferred' object is not subscriptable

### ISSUE THREAD

**Issue by karthic23** [sbobovyc/GameTools#16]
Hi Having an issue when importing my CSV into blender, I have exported 'VS in' From Renderdoc, however getting an error shown below. I have uploaded the CSV file to [https://we.tl/t-soQgmYMG3Y].

Python: Traceback (most recent call last):
  File "C:\Users\XXX\AppData\Roaming\Blender Foundation\Blender\2.93\scripts\addons\Blender_Import_CSV.py", line 91, in execute
    global_matrix = axis_conversion(from_forward = self.axis_forward, from_up = self.axis_up).to_4x4()
  File "D:\Program Files\Blender\2.93\scripts\modules\bpy_extras\io_utils.py", line 302, in axis_conversion
    if from_forward[-1] == from_up[-1] or to_forward[-1] == to_up[-1]:
TypeError: '_PropertyDeferred' object is not subscriptable

location: <unknown location>:-1

Appreciate it very much if you could help.

Kind Regards.

**Comment by sbobovyc** [comment]
The script you are using was written and extensively used with Blender 2.8x. Blender 2.9x introduced many breaking changes. I suggest you use this script https://github.com/sbobovyc/GameTools/blob/master/common/raw2obj.py to convert your CSV into an OBJ and import that into Blender. Let me know if that works for you.

**Comment by Hopefullyidontgetbanned** [comment]
`python raw2obj.py --type positions --input-type 'csv' Nanashi_0.csv >> nanashi.obj`

Example of how i did it with Nanashi from Shin Megami Tensei IV Apocalypse using citra

**Comment by karthic23** [comment]
Hi Thanks guys! Worked perfectly, Just wondering if it were at all possible to import the obj's to the location from the extracted game so when I import multiple objs to blender they are in correct relation to each other.

**Comment by sbobovyc** [comment]
Objects are transformed by the Model-View-Projection (MVP) matrix from object space to camera space. If you could get your hands on the Model and View matrices, then you could apply them to the object vertices to get what you are asking. How and where the MVP is computed is application specific. It can be done by the shader or in the application. Nvidia has a tool called Ansel that captures a 3D "screenshot", but I believe it has to be integrated into the application. 

TLDR It is possible, but not without a lot of effort.

**Comment by karthic23** [comment]
Thank you very much I will look into and try get it to work.  Thank you for your quick responses and help.

**Comment by sbobovyc** [comment]
Glad to help!

**Comment by bonkmaykrQ** [comment]
Hello--sorry to necro this thread, but I was having the same issue so I tried using raw2ob.py as suggested, and it threw this:
```
Traceback (most recent call last):
  File "C:\Users\kangang\Downloads\raw2obj.py", line 23, in <module>
    import pandas as pd
ImportError: No module named pandas
```

Am I missing some libraries that I'm supposed to have installed? Or would it just be easier to downgrade to Blender 2.8?

**Comment by sbobovyc** [comment]
You will need the pandas library: https://pandas.pydata.org/

I typically leave the Blender Python install as is and use a separate install for these scripts. You have many options of how to install Python on Windows.
